jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -1048,6 +1057,45 @@ public static void setProperties(Properties p) {
 	}
 }
 
+static void initSecurityManager(ClassLoader applicationClassLoader) {
+	String javaSecurityManager = internalGetProperties().getProperty("java.security.manager"); //$NON-NLS-1$
+
+	/*[IF JAVA_SPEC_VERSION >= 12]*/
+	if ("disallow".equals(javaSecurityManager) //$NON-NLS-1$
+	/*[IF JAVA_SPEC_VERSION >= 18]*/
+	|| (javaSecurityManager == null)
+	/*[ENDIF] JAVA_SPEC_VERSION >= 18 */
+	) {
+		throwUOEFromSetSM = true;
+	}
+	/*[ENDIF] JAVA_SPEC_VERSION >= 12 */
+
+	if ((javaSecurityManager != null)
+	/*[IF JAVA_SPEC_VERSION >= 12]*/
+	/* See the SecurityManager javadoc for details about special tokens. */
+	&& !javaSecurityManager.equals("disallow") //$NON-NLS-1$ /* special token to disallow SecurityManager */
+	&& !javaSecurityManager.equals("allow") //$NON-NLS-1$ /* special token to allow SecurityManager */
+	/*[ENDIF] JAVA_SPEC_VERSION >= 12 */
+	) {
+		/*[IF JAVA_SPEC_VERSION >= 17]*/
+		initialErr.println("WARNING: A command line option has enabled the Security Manager"); //$NON-NLS-1$
+		initialErr.println("WARNING: The Security Manager is deprecated and will be removed in a future release"); //$NON-NLS-1$
+		/*[ENDIF] JAVA_SPEC_VERSION >= 17 */
+		if (javaSecurityManager.isEmpty() || "default".equals(javaSecurityManager)) { //$NON-NLS-1$
+			setSecurityManager(new SecurityManager());
+		} else {
+			try {
+				Constructor<?> constructor = Class.forName(javaSecurityManager, true, applicationClassLoader).getConstructor();
+				constructor.setAccessible(true);
+				setSecurityManager((SecurityManager)constructor.newInstance());
+			} catch (Throwable e) {
+				/*[MSG "K0631", "JVM can't set custom SecurityManager due to {0}"]*/
+				throw new Error(com.ibm.oti.util.Msg.getString("K0631", e.toString()), e); //$NON-NLS-1$
+			}
+		}
+	}
+}
+
```

Comment:
How about following code snippet?
```
	/*[IF JAVA_SPEC_VERSION > 11]*/
	if ("allow".equals(javaSecurityManager)){  //$NON-NLS-1$
		// do nothing for the special token to allow SecurityManager
	} else if ("disallow".equals(javaSecurityManager) //$NON-NLS-1$
		/*[IF JAVA_SPEC_VERSION >= 18]*/
		|| (javaSecurityManager == null)
		/*[ENDIF] JAVA_SPEC_VERSION >= 18 */
	) {
		throwUOEFromSetSM = true;
	} else
	/*[ENDIF] JAVA_SPEC_VERSION > 11 */
	if (javaSecurityManager != null) {
        ....
```

Reply:
Updated. `java.security.manager` system property was introduced in JDK12. 
- `java.security.manager` does not exist in [JDK11  java/lang/SecurityManager Javadoc](https://cr.openjdk.java.net/~iris/se/11/spec/latest/api/java.base/java/lang/SecurityManager.html#java.security.manager).
- `java.security.manager` exists in [JDK12 java/lang/SecurityManager Javadoc](https://cr.openjdk.java.net/~iris/se/12/spec/latest/api/java.base/java/lang/SecurityManager.html#java.security.manager).

JDK8 and JDK11 do not need to recognize the `java.security.manager` system property. Hence, I have enabled the code related to `java.security.manager` only for `JAVA_SPEC_VERSION > 11`.

Reply:
Although `java.security.manager` is not documented, I found that it is still recognized by JDK8 and JDK11. So, I have updated the PR as per the initial suggested in https://github.com/eclipse-openj9/openj9/pull/14329#discussion_r791161285. I have reverted my changes described in https://github.com/eclipse-openj9/openj9/pull/14329#discussion_r791241857.

Reply:
It appears not in Java doc, but in web server application doc such as https://www.ibm.com/docs/en/cics-ts/4.2?topic=jvm-enabling-java-security-manager and a few others.